pkg/destroy/gcp/address.go

```diff
@@ -3,27 +3,43 @@ package gcp
 import (
 	"context"
 	"fmt"
+	"strings"
 
-	"github.com/sirupsen/logrus"
+	"github.com/pkg/errors"
 	"google.golang.org/api/compute/v1"
 	"google.golang.org/api/googleapi"
 
 	"github.com/openshift/installer/pkg/types/gcp"
 )
 
-func (o *ClusterUninstaller) listAddresses(ctx context.Context, scope resourceScope) ([]cloudResource, error) {
-	return o.listAddressesWithFilter(ctx, "items(name,region,addressType),nextPageToken", o.clusterIDFilter(), nil, scope)
+func (o *ClusterUninstaller) listAddresses(ctx context.Context, typeName string, listFunc addressListFunc) ([]cloudResource, error) {
+	return o.listAddressesWithFilter(ctx, typeName, "items(name),nextPageToken", o.clusterIDFilter(), listFunc)
 }
 
-func createAddressCloudResources(filterFunc func(address *compute.Address) bool, list *compute.AddressList) []cloudResource {
+// listAddressesWithFilter lists addresses in the project that satisfy the filter criteria.
+// The fields parameter specifies which fields should be returned in the result, the filter string contains
+// a filter string passed to the API to filter results. The filterFunc is a client-side filtering function
+// that determines whether a particular result should be returned or not.
```

Comment:
These comment docs need to be rewritten. There is no `filterFunc`, only `listFunc`.